src/main/java/com/coveo/pushapiclient/PushSource.java

```diff
@@ -0,0 +1,300 @@
+     *
+     */
+    public PushSource(String apiKey, String organizationId, String sourceId) {
+        PlatformUrl platformUrl = new PlatformUrl(PlatformUrl.DEFAULT_ENVIRONMENT, PlatformUrl.DEFAULT_REGION);
+        this.sourceId = sourceId;
+        this.platformClient = new PlatformClient(apiKey, organizationId,
+                platformUrl);
+    }
+
+    /**
+     * Create a Push source instance from its
+     * <a href="https://docs.coveo.com/en/3295#stream-api-url">Stream API URL</a>
+     *
+     * @param apiKey         The API key used for all operations regarding your
+     *                       source.
+     *                       <p>
+     *                       Ensure your API key has the required privileges for the
+     *                       operation you will be performing
+     *                       *
+     *                       <p>
+     *                       For more information about which privileges are
+     *                       required,
+     *                       see
+     *                       <a href=
+     *                       "https://docs.coveo.com/en/1707#sources-domain">Privilege
+     *                       Reference.</a>
+     *
+     * @param organizationId The unique identifier of your organization.
+     *                       <p>
+     *                       The Organization Id can be retrieved in the URL of your
+     *                       Coveo organization.
+     *
+     * @param sourceId       The unique identifier of the target Push source.
+     *                       <p>
+     *                       The Source Id can be retrieved when you edit your
+     *                       source in the <a href=
+     *                       "https://docs.coveo.com/en/183/glossary/coveo-administration-console">Coveo
+     *                       Administration Console</a>
+     *
+     * @param platformUrl    The object containing additional information on the
+     *                       URL endpoint.
+     *                       You can use the {@link PlatformUrl} when your
+     *                       organization is located in a non-default Coveo
+     *                       environement and/or region.
+     *
+     */
+    public PushSource(String apiKey, String organizationId, String sourceId, PlatformUrl platformUrl) {
+        this.sourceId = sourceId;
+        this.platformClient = new PlatformClient(apiKey, organizationId,
+                platformUrl);
+    }
+
+    /**
+     * Create or update a security identity. See [Adding a Single Security
+     * Identity](https://docs.coveo.com/en/167) and [Security Identity
+     * Models](https://docs.coveo.com/en/139).
+     *
+     * @param securityProviderId
+     * @param securityIdentityModel
+     * @return
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    public HttpResponse<String> createOrUpdateSecurityIdentity(String securityProviderId,
```

Comment:
What does this `String` mean? Is this the type of the body?

If creating/updating the security identity fails, does this method throw, or does it return a HttpResponse?

Reply:
> What does this String mean? Is this the type of the body?
Yes, but we could map the body response to the appropriate type in a future improvement.

> If creating/updating the security identity fails, does this method throw, or does it return a HttpResponse?
return an HttpResponse with a non 200 status code